File: 백준/Silver/1874. 스택 수열/스택 수열.cc
#include<iostream>
#include<stack>
#include<vector>
#pragma warning (disable:4996)

using namespace std;

stack<int> stk;
vector<char> v;

int n;
int recent = 1, input;

int main()
{
	scanf("%d", &n);

	for (int i = 1; i <= n; i++) {
		scanf("%d", &input);
		while (recent <= input) {
			stk.push(recent++);
			v.push_back('+');
		}
		if (input == stk.top()) {
			stk.pop();
			v.push_back('-');
		}
		else {
			printf("NO");
			return 0;
		}
	}
	for (int i = 0; i < v.size(); i++) {
		printf("%c\n", v[i]);
	}
}

File: 백준/Silver/1874. 스택 수열/스택 수열.java

import java.util.*;
import java.io.*;


public class Main {

    int n,num,recent=1;

    BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
    StringTokenizer st;
    Stack<Integer> stk = new Stack<>();

    void solution() throws Exception {
        n = Integer.parseInt(br.readLine());

        StringBuilder sb = new StringBuilder();

        for(int i=1;i<=n;i++){
            num = Integer.parseInt(br.readLine());
            while(recent<=num){
                stk.push(recent++);
                sb.append("+").append("\n");
            }
            if(num == stk.peek()){
                stk.pop();
                sb.append("-").append("\n");
            }
            else{
                System.out.println("NO");
                return;
            }
        }
        System.out.println(sb);
    }



    public static void main(String[] args) throws Exception {
       new Main().solution();
    }
}